app/Http/Controllers/PostsController.php

```diff
@@ -49,12 +50,32 @@ public function store(Request $request)
     {
         $this->validate($request,[
             'title'=>'required',
-            'body'=>'required'
+            'body'=>'required',
+            'cover_image'=>'image|nullable|max:1999'
         ]);
+
+        //Handle File uploading 
+        if ($request->hasfile('cover_image')){
+            //Get file name with the extension
+            $fileNameWithExt = $request->file('cover_image')->getClientOriginalName();
+            //Get just file name
+            $fileName = pathinfo($fileNameWithExt, PATHINFO_FILENAME);
```

Comment:
This part is a straight up PHP thing to get file name. Instructor said there is no Laravel way to do so... is that right???